project_group_setup.py

```diff
@@ -0,0 +1,270 @@
+    osg_co_id = OSG_CO_ID
+    ucid = UNIX_CLUSTER_ID
+    provision_target = LDAP_TARGET_ID
+    outfile = None
+    authstr = None
+    ldap_authtok = None
+    project_gid_startval = PROJECT_GIDS_START
+
+
+options = Options()
+
+
+def parse_options(args):
+    try:
+        ops, args = getopt.getopt(args, "u:c:g:l:p:d:f:e:o:h")
+    except getopt.GetoptError:
+        usage()
+
+    if args:
+        usage("Extra arguments: %s" % repr(args))
+
+    passfd = None
+    passfile = None
+
+    for op, arg in ops:
+        if op == "-h":
+            usage()
+        if op == "-u":
+            options.user = arg
+        if op == "-c":
+            options.osg_co_id = int(arg)
+        if op == "-g":
+            options.ucid = int(arg)
+        if op == "-l":
+            options.provision_target = int(arg)
+        if op == "-p":
+            options.ldap_authtok = arg
+        if op == "-d":
+            passfd = int(arg)
+        if op == "-f":
+            passfile = arg
+        if op == "-e":
+            options.endpoint = arg
+        if op == "-o":
+            options.outfile = arg
+
+    try:
+        user, passwd = utils.getpw(options.user, passfd, passfile)
+        options.authstr = utils.mkauthstr(user, passwd)
+    except PermissionError:
+        usage("PASS required")
+
+
+def append_if_project(project_groups, group):
+    # If this group has a ospoolproject id, and it starts with "Yes-", it's a project
+    if utils.identifier_matches(group["ID_List"], "ospoolproject", (OSPOOL_PROJECT_PREFIX_STR + "*")):
+        # Add a dict of the relavent data for this project to the project_groups list
+        project_groups.append(group)
+
+
+def update_highest_osggid(highest_osggid, group):
+    # Get the value of the osggid identifier, if this group has one
+    osggid = utils.identifier_from_list(group["ID_List"], "osggid")
+    # If this group has a osggid, keep a hold of the highest one we've seen so far
+    if osggid is not None:
+        return max(highest_osggid, int(osggid))
+    else:
+        return highest_osggid
+
+
+def get_comanage_data():
+    projects_list = []
+    highest_osggid = 0
+
+    co_groups = utils.get_osg_co_groups(options.osg_co_id, options.endpoint, options.authstr)["CoGroups"]
+    for group_data in co_groups:
+        try:
+            identifier_list = utils.get_co_group_identifiers(group_data["Id"], options.endpoint, options.authstr)
+            # Store this groups data in a dictionary to avoid repeated API calls
+            group = {"Gid": group_data["Id"], "Name": group_data["Name"], "ID_List": identifier_list["Identifiers"]}
+
+            # Add this group to the project list if it's a project, otherwise skip.
+            append_if_project(projects_list, group)
+
+            # Update highest_osggid, if this group has an osggid and it's higher than the current highest osggid.
+            highest_osggid = update_highest_osggid(highest_osggid, group)
+        except TypeError:
+            pass
```

Comment:
What causes a `TypeError` here? Can we dump a message in case this sort of thing is relevant?

Reply:
This used to be an if statement that checked to see if identifier_list was not None before checking to see if it had a Project identifier, now it just passes if the identifier_list is None because an empty list can't contain the identifier we're looking for.